toolchain/lex/tokenized_buffer.cpp

```diff
@@ -49,20 +55,35 @@ auto VariantMatch(V&& v, Fs&&... fs) -> decltype(auto) {
   return std::visit(Overload{std::forward<Fs&&>(fs)...}, std::forward<V&&>(v));
 }
 
-#if __x86_64__
-#define CARBON_USE_SIMD 1
-
+#if CARBON_USE_SIMD
 // A table of masks to include 0-16 bytes of an SSE register.
-static const std::array<__m128i, sizeof(__m128i) + 1> prefix_masks = [] {
-  std::array<__m128i, sizeof(__m128i) + 1> masks = {};
-  for (auto [i, mask] : llvm::enumerate(masks)) {
-    memset(&mask, 0xFF, i);
-  }
-  return masks;
-}();
+static const
+#if __ARM_NEON
+    std::array<uint8x16_t, sizeof(uint8x16_t) + 1>
+#elif __x86_64__
+    std::array<__m128i, sizeof(__m128i) + 1>
 #else
-#define CARBON_USE_SIMD 0
+#error "Unsupported SIMD architecture!"
+#endif
+        prefix_masks = [] {
+          using MaskArrayT = std::remove_const_t<decltype(prefix_masks)>;
+          using MaskT = MaskArrayT::value_type;
+          MaskArrayT masks = {};
```

Comment:
Had you considered moving the `using` into the platform `#if`, removing the `remove_const_t`, e.g.:

```
#if _ARM_NEON
using MaskT = uint8x16_t;
#elif __x86_64__
using MaskT = __m128i;
#else
#error "Unsupported SIMD architecture!"
#endif
using MaskArrayT = std::array<MaskT, sizeof(MaskT) + 1>;
static const MaskArrayT prefix_masks = [] {
  MaskArrayT masks = {};
```

Reply:
Sure, done. Had to add an anonymous namespace and change the names to make more sense in the broader scope, but seems fine and especially with a portable way of building the mask seems pretty clean.